### PR TITLE
Added the functionality for disabling the copy and download buttons when nothing to preview

### LINKED ISSUE #8: When the "Nothing to preview" message is displayed, buttons should be disabled
As the title says, the buttons should be disabled and not clickable when there's no data there to preview.

image

### PR DESCRIPTION
**Pull request by masterghost2002**
## issue

fixes #8

### CODE REVIEW

**Review by allejo — CHANGES_REQUESTED** [review]
Very close! The problem here is that the "Nothing to preview yet!" message shows up when `processed.length <= 1` (the header row counts as 1). Your PR allows for the buttons to show up when there's nothing to preview. I would add a `useEffect` to listen to when `processed` changes and *that's* when `setShowButtons` should be called.

image

**Comment by masterghost2002** [comment]
Thank you for the guidance sir,
here is a quick view
[Screencast from 2023-09-02 14-55-21.webm](https://github.com/allejo/json-reshaper/assets/55751461/c75793bb-2c13-4c7f-bab6-f2b534490023)

File: src/components/TransformDelimiterSeparatedValues.tsx [hunk 1]
```diff
@@ -41,12 +43,15 @@ export const TransformDelimiterSeparatedValues = ({
 	);
 	const exportToDsv = useCallback(() => generatedDSV, [generatedDSV]);
 
+	useEffect(()=>{
+		setShowButtons(()=>processed.length>1);
```
Comment (allejo) [inline src/components/TransformDelimiterSeparatedValues.tsx:47]:
```suggestion
		setShowButtons(processed.length > 1);
```

I don't believe you need this to be a function call, it'll evaluate to a boolean by itself. It only makes sense to use a function in a state setter if the previous state is needed

File: src/components/TransformDelimiterSeparatedValues.tsx [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { applyReshapeTransformationArray } from '../utilities.ts';
 
 interface Props extends IOutputComponentProps {
 	delimiter: string;
+	setShowButtons:React.Dispatch<React.SetStateAction<boolean>>;
```
Comment (allejo) [inline src/components/TransformDelimiterSeparatedValues.tsx:8]:
```suggestion
	setShowButtons: StateSetter<boolean>;
```

I have an alias for this type

**Review by allejo — APPROVED** [review]
I'm going to merge this PR manually because I made some formatting changes. A few notes for you to take away (in addition to the comments I left below)

1. When you make a PR, never make a PR against your default branch (i.e. `main` or `master`). For example, in this PR you made the PR from `masterghost2002:main` to `allejo:main`; this causes problems like not allowing project maintainers to push to your branch
2. When you make a PR that directly fixes an issue, make sure to use a [closing keyword](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue) so that the PR will automatically mark the issue as completed when it's merged. Example closing keywords are, "fixes," "fixed," "closed," etc.

**Pull request closed without merging**